Source/Core/DolphinQt/QtUtils/DoubleClickEventFilter.cpp

```diff
@@ -3,12 +3,69 @@
 
 #include "DolphinQt/QtUtils/DoubleClickEventFilter.h"
 
+#include <QChildEvent>
 #include <QEvent>
+#include <QMouseEvent>
 
-bool DoubleClickEventFilter::eventFilter(QObject* object, QEvent* event)
+DoubleClickEventFilter::DoubleClickEventFilter(const int valid_button_mask, QObject* const parent)
+    : QObject(parent), m_valid_button_mask(valid_button_mask)
 {
-  if (event->type() == QEvent::MouseButtonDblClick)
-    emit doubleClicked();
```

Comment:
Couldn't we just check a button mask here and connect actions to the `&DoubleClickEventFilter::doubleClicked` signal for the game list/grid, like was already being done for the "Select a Directory" widget? All this filter installation on descendants just seems like a lot.

Reply:
That wouldn't work because, unlike for the `m_empty` (Select a Directory) widget, DoubleClickEventFilter never receives the event when applied to `m_list` or `m_grid`; the event gets consumed by one of the child widgets, while `m_empty` is just a QLabel with no children.

Fortunately I found a better way to get the event, so we can skip the filter installations anyway.